server/job_service.go

```diff
@@ -254,36 +259,11 @@ func (s *jobService) StopJob(channelID string) error {
 }
 
 func (s *jobService) GetJob(jobID string) (job.Job, error) {
-	jb, err := s.client.GetJob(jobID)
-
-	// Adding a check in case the service restarted and lost credentials. This is
-	// a common case when the offloader is running in kubernetes deployment. The
-	// solution is to re-initialize the service which will cause a new registration
-	// attempt.
-	if errors.Is(err, offloader.ErrUnauthorized) {
-		if err := s.reinitializeOnAuthError(); err != nil {
-			return job.Job{}, err
-		}
-		return s.client.GetJob(jobID)
-	}
-	return jb, err
```

Comment:
Technically we could use `retryJobService` here but we are likely going to deprecate this API given it won't work in K8S or any scenario involving load balancing of multiple offloader instances.